### PR TITLE
ListObjects V2

### PR DESCRIPTION
**Pull request by DPG17**
Developed system tests(ST) and UT

### CODE REVIEW

**Comment by DPG17** [comment]
Jenkins build is successful:
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/S3server/37

File: st/clitests/awss3api_spec.py [hunk 1]
```diff
@@ -99,6 +134,69 @@ def delete_object_list_file(file_name):
...
+lv2_response = get_aws_cli_object(result.status.stdout)
+# Get next-token from lv2_response
+if ("next_token" in lv2_response.keys()):
+    Next_token = lv2_response["next_token"]
+
+# Get keys from lv2_response
+obj_keys =  lv2_response["keys"]
+
+# Assert for key existence
+assert "foo" in obj_keys, "Key \"foo\" is expected to be in the result"
```
Comment (amitwac2608) [inline st/clitests/awss3api_spec.py:163]: should we add negative test also :- 

assert key not in obj_keys ?

File: ut/s3_common_utilities_test.cc [hunk 2]
```diff
@@ -276,3 +276,12 @@ TEST_F(S3CommonUtilitiesTest, IsYamlValueNullTest6) {
...
   EXPECT_FALSE(S3CommonUtilities::is_yaml_value_null(input_string));
 }
+
+TEST_F(S3CommonUtilitiesTest, VerifyXORObfuscation) {
+  std::shared_ptr<S3CommonUtilities::S3Obfuscator> pXORObfuscator =
+      std::make_shared<S3CommonUtilities::S3XORObfuscator>();
+  std::string input = "This is to be obfuscated. Check de-obfuscation";
+  std::string strObfuscated = pXORObfuscator->encode(input);
+  std::string output = pXORObfuscator->decode(strObfuscated);
+  EXPECT_TRUE(input == output);
```
Comment (amitwac2608) [inline ut/s3_common_utilities_test.cc:286]: I am not sure if this suggestion would be better, but it we want to see to strings equal or not, we can use EXPECT_STREQ

**Review by amitwac2608 — APPROVED** [review]
looks good

**Review by knrajnambiar76 — APPROVED** [review]

File: ut/s3_common_utilities_test.cc (hunk 2 above)
Comment (DPG17) [reply ut/s3_common_utilities_test.cc:286]: yes, can be done. But, at this point, will keep code unchanged.

File: st/clitests/awss3api_spec.py (hunk 1 above)
Comment (DPG17) [reply st/clitests/awss3api_spec.py:163]: Will try to add -ve test case

Comment (DPG17) [reply st/clitests/awss3api_spec.py:162]: _ve test case added below.

**Comment by DPG17** [comment]
Build success:  http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/S3server/50/
Build on commit: 74457f75282bafd93f35adcae305ddd56ce70038

**Pull request merged**